dotnet/src/Experimental/Process.Runtime.Dapr/Actors/StepActor.cs

```diff
@@ -390,12 +399,21 @@ internal async ValueTask EmitEventAsync(ProcessEvent daprEvent)
         }
 
         // Get the edges for the event and queue up the messages to be sent to the next steps.
-        foreach (var edge in this.GetEdgeForEvent(daprEvent.Id!))
+        bool foundEdge = false;
+        foreach (var edge in this.GetEdgeForEvent(daprEvent.Id))
         {
             ProcessMessage message = ProcessMessageFactory.CreateFromEdge(edge, daprEvent.Data);
             var scopedStepId = this.ScopedActorId(new ActorId(edge.OutputTarget.StepId));
             var targetStep = this.ProxyFactory.CreateActorProxy<IMessageBuffer>(scopedStepId, nameof(MessageBufferActor));
             await targetStep.EnqueueAsync(message).ConfigureAwait(false);
+            foundEdge = true;
+        }
+
+        // Error event was raised with no edge to handle it, send it to the global error event buffer.
+        if (!foundEdge && daprEvent.IsError)
+        {
+            var parentProcess1 = this.ProxyFactory.CreateActorProxy<IEventBuffer>(new ActorId(ProcessConstants.GlobalErrorEventId), nameof(EventBufferActor));
```

Comment:
Should this event buffer be scoped to the process also?

Reply:
Yeah, it needs to match the input pattern. I'll update both.

Reply:
Updated